File: binary-search/1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp;
        int pass, n = nums.size();
        
        for(int i = 0; i < n; i++){
            pass = target - nums[i];
            if(mp.find(pass) != mp.end()){
                return {mp[pass], i};
            }
            mp[nums[i]] = i;
        }
        return {};
        
    }
};

File: array/1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> pairMap = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];

            if (pairMap.containsKey(target - num)) {
                return new int[] { i, pairMap.get(target - num) };
            }
            pairMap.put(num, i);
        }
        return new int[] {};
    }
}